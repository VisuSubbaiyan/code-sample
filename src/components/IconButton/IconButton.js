import React from 'react';
import PropsTypes from 'prop-types';

const IconButton = props => (
  <div onClick={props.onClick}>
    <i className={`fa fa-${props.icon}`}></i>
  </div>
);

IconButton.propTypes = {
  icon: PropsTypes.string.isRequired
}

export default IconButton;
