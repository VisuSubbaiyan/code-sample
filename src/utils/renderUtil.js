import React from 'react';

import {Airport, BoardingPass, Ground} from '../components';

export const renderItems = items => items.map(item => {
  const { type, segment_id } = item;

  switch (type) {
    case 'GROUND': {
      return <Ground {...item} key={segment_id} className="segment-active no-top-padding" />;
    }
    
    case 'AIRPORT': {
      return <Airport {...item} key={segment_id} />;
    }

    default:
      return <BoardingPass {...item} key={segment_id} />
  }
});
