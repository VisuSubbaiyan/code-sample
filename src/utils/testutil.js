import { mount, shallow } from 'enzyme';
import { prettyPrint } from 'html';

export const mockFunction = jest.fn(value => value);

export const sampleFriend = {
  name: 'Sample friend',
  gender: 'male'
};

export const sampleId = 1;

export const createMountedComponent = ({ children, isShallow }) => {
  const mountedComponent = isShallow ? shallow(children) : mount(children);
  mountedComponent.htmlRender = () => prettyPrint(mountedComponent.html(), { indent_size: 2 });

  return mountedComponent;
};

