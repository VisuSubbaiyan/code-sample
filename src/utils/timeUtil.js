import moment from 'moment';

export const formatTime = time => {
  const tempTime = moment.duration(time);
  
  return `${tempTime.hours()}: ${tempTime.minutes()}`;
}