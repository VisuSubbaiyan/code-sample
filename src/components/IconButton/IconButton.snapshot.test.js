import React from 'react';

import { createMountedComponent } from '../../utils/testutil';
import IconButton from './IconButton';

describe('<Header /> should', () => {
  const mountedComponent = createMountedComponent({
    children: <IconButton icon='bus' />
  });

  const icon = mountedComponent.find('i');

  it('render correctly according to props', () => {
    expect(mountedComponent.htmlRender()).toMatchSnapshot();
    expect(icon).toHaveLength(1);
  })
});