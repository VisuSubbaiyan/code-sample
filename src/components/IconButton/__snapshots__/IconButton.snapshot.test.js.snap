// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render correctly according to props 1`] = `
"<div><i class=\\"fa fa-bus\\"></i>
</div>"
`;
